test-engineering/client/tests/test_merino.py

```diff
@@ -40,7 +41,11 @@ def test_merino(merino_url: str, steps: List[Step]):
         if r.status_code == 200:
             # If the response status code is 200 OK, load the response content
             # into a Python dict and generate a dict from the response model
-            assert r.json() == step.response.content.dict()
+            merino_response = r.json()
+            request_id = merino_response.pop("request_id")
+            assert UUID(request_id)
+            assert merino_response == step.response.content.dict()
```

Comment:
What do you mean?

Reply:
oh I mean like i popped out of the response dict before comparing it to `step.response.content.dict()` I wasn't sure if there was a better way of doing this, since request_ids are assigned randomly and changes each time? so I couldn't hard code anything to `scenarios.yml`?

Reply:
Ohh. Yeah this seems good. I think the other way would be something like a [`Whatever`](https://github.com/mozilla/normandy/blob/e5f3b8f7f061729da593130afc3ac6111a1eaa21/normandy/base/tests/__init__.py#L12-L51) helper that is always equal to everything. Seems like overkill though.

Reply:
I think pydantic's features are helpful for this: We can add the `request_id` as a required field with type `UUID` to the model, to verify that responses from Merino include that field and to also check that it's a valid UUID. But then exclude the `request_id` field when exporting the model to a dict when comparing against the expected value from `scenarios.yml`.

The error message for when Merino's response does **not** include `response_id` would look like this:
```text
>   ???
E   pydantic.error_wrappers.ValidationError: 1 validation error for ResponseContent
E   request_id
E     field required (type=value_error.missing)
```

The error message for when the value for  `request_id` in Merino's response is not a valid UUID would look like this:
```text
>   ???
E   pydantic.error_wrappers.ValidationError: 1 validation error for ResponseContent
E   request_id
E     value is not a valid uuid (type=type_error.uuid)
```

Do you mind if I push a commit with these changes to this branch @tiftran?